lib/helpers/getSpecificationObject.js

```diff
@@ -0,0 +1,71 @@
+const parser = require('swagger-parser');
+const createSpecification = require('./createSpecification');
+const specHelper = require('./specification');
+const parseApiFile = require('./parseApiFile');
+const filterJsDocComments = require('./filterJsDocComments');
+const convertGlobPaths = require('./convertGlobPaths');
+
+function isEmptyObject(obj) {
+  // eslint-disable-next-line
+  Object.keys(obj).forEach(key => {
+    if (key in obj) return false;
+  });
+
+  return true;
+}
+
+/**
+ * OpenAPI specification validator does not accept empty values for a few properties.
+ * Solves validator error: "Schema error should NOT have additional properties"
+ * @function
+ * @param {object} inputSpec - The swagger/openapi specification
+ * @param {object} improvedSpec - The cleaned version of the inputSpec
+ */
+function cleanUselessProperties(specification) {
+  const spec = JSON.parse(JSON.stringify(specification));
```

Comment:
Why stringify then parse ?

Params in the jsdoc comment are not correct.

Reply:
This is a typical pattern in JS for cloning entities recursively.

Reply:
@kibertoad is right. Deep clone without reaching for a third party library. That's one of the very few places where I reached for changing the previous logic of mutating objects. There are many other places where this change will make the library better, but I didn't want to go deep in the rabbit hole changing too much unrelated code.